src/msg.rs

```diff
@@ -357,6 +362,15 @@ impl QueryMsg {
     }
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum QueryWithPermit {
+    Allowance { spender: HumanAddr },
+    Balance {},
+    TransferHistory { page: Option<u32>, page_size: u32 },
+    TransactionHistory { page: Option<u32>, page_size: u32 },
+}
```

Comment:
I don't love that this is a separate enum... It makes it so that new queries have to be duplicated...
Going forward i think we can solve this by changing our conventions around the structure of the QueryMessage itself. Basically, make queries that need VKs to look like:
```rs
enum QueryMessage {
  ...
  WithKey {
    key: String,
    query: AuthenticatedQuery,
  }
}
```
to match `WithPermit`, and then both can use the same format of query. but i guess now we're stuck with some legacy patterns?

In any case, can we rename this type to `AuthenticatedQuery`?

Reply:
`Permit` is the term for this in Ethereum.